cpp/src/arrow/c/bridge.cc

```diff
@@ -1601,6 +1621,23 @@ struct ArrayImporter {
     return Status::OK();
   }
 
+  Status Visit(const RunEndEncodedType& type) {
+    RETURN_NOT_OK(CheckNumChildren(2));
+    if (c_struct_->n_buffers == 1) {
+      // REE arrays are not required to have any buffer,
+      // but an empty validity bitmap buffer is allowed.
+      RETURN_NOT_OK(AllocateArrayData());
```

Comment:
I initially missed this, but is there a reason we want to allow this? We don't have any legacy implementations to worry about for this new datatype.

Reply:
The current implementation itself only works by always having the first buffer set to null because so much code assumes `buffers[0]` to be checkable. @bkietz even pinged me about some issues he had when this wasn't respected.

Reply:
We should distinguish between what we're doing internally in Arrow C++ and what we export/import using the C Data Interface. I'm not sure accepting a buffer for REE arrays makes sense if we don't export such a buffer.

Reply:
Agreed, I wouldn't expect any exporters of REE arrays to emplace the placeholder null buffer. Let's remove the above check `if (c_struct_->n_buffers == 1) {` and require zero buffers

Reply:
Done.